c-api.rst

```diff
@@ -101,8 +101,44 @@ No changes that break the Stable ABI are allowed.
 The Limited API should be defined in ``Include/``, excluding the
 ``cpython`` and ``internal`` subdirectories.
 
-Guidelines for changing the Limited API
----------------------------------------
+
+Guidelines for changing the Limited API, and removing items from it
+-------------------------------------------------------------------
+
+While the *Stable ABI*  must not be broken, the existing Limited API can be
+changed, and items can be removed from it, if:
+
+- the Backwards Compatibility Policy (:pep:`387`) is followed, and
+- the Stable ABI is not broken -- that is, extensions compiled with
+  Limited API of older versions of Python continue to work on
+  newer versions of Python.
```

Comment:
Is it possible to export a function which always fail, if the function has been removed? Removing PyUnicode_InternImmortal() sounds like an interesting challenge: https://bugs.python.org/issue41692

Maybe the function can be replaced with a function... which does nothing but log an error?

Reply:
> Is it possible to export a function which always fail, if the function has been removed?

I believe, only if the function is not vital for already compiled modules (so its failure won't hamper their main activity). That's actually the whole point of backwards compatibility, to not break third party things that already work.

Reply:
IMO, after it's removed from the API *and* when `SSTATE_INTERNED_IMMORTAL` needs to be removed, `PyUnicode_InternImmortal` can do an unpaired INCREF: make the string immortal, but leak it.
See https://www.python.org/dev/peps/pep-0652/#stable-abi:

> Future Python versions may deprecate some members of the Stable ABI. Deprecated members will still work, but may suffer from issues like reduced performance or, in the most extreme cases, memory/resource leaks.